### PR TITLE
MWPW-129660 Override Marketo Form Params

### PR DESCRIPTION
**Pull request by Brandon32**
* Allow setting or overriding specific from configurations.
* Decorating destination URL for form success redirect.

Resolves: [MWPW-129660](https://jira.corp.adobe.com/browse/MWPW-129660)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/bmarshal/mcz/marketo-block?martech=off
- After: https://bmarshal-marketo-overrides--milo--adobecom.hlx.page/drafts/bmarshal/mcz/marketo-block?martech=off

**BACOM URLs:**
- Before: https://main--bacom--adobecom.hlx.live/uk/resources/reports/state-of-marketing-automation-b?martech=off
- After: https://main--bacom--adobecom.hlx.live/uk/resources/reports/state-of-marketing-automation-b?milolibs=bmarshal-marketo-overrides&martech=off

### CODE REVIEW

File: libs/blocks/marketo/marketo.js
Comment (meganthecoder) [inline libs/blocks/marketo/marketo.js:22]: Unnecessary extra line
```suggestion
const MUNCHKIN_ID = 'marketo munckin';
const FORM_MAP = {
```

File: libs/blocks/marketo/marketo.js
Comment (Brandon32) [reply libs/blocks/marketo/marketo.js:115]: Yes that is redundant, thank you.

File: libs/blocks/marketo/marketo.js
Comment (mokimo) [inline libs/blocks/marketo/marketo.js:24]: These look like very odd strings, but I guess that's why this map exist

**Review by mokimo — APPROVED** [review]

File: libs/blocks/marketo/marketo.js
Comment (Brandon32) [reply libs/blocks/marketo/marketo.js:24]: I have created an authoring guide https://milo.adobe.com/docs/authoring/marketo.
We are converting author friendly names to values that are added to `window.mcz_marketoForm_pref`

Comment (mokimo) [reply libs/blocks/marketo/marketo.js:24]: Since those are specific strings and the spaces etc. need to be set, maybe `destination-url` `co-partner-names` and `sfdc-campaign-id` ? 

Not blocking the PR, but it might be more obvious it's something "pasteable" to authors then, but I'll leave it up to you

Comment (Brandon32) [reply libs/blocks/marketo/marketo.js:24]: @mokimo That's a good idea, I'll string replace spaces with dashes so we're more flexible with authoring.

File: test/blocks/marketo/marketo.test.js
Comment (JasonHowellSlavin) [inline test/blocks/marketo/marketo.test.js:103]: super nit `decorates`

**Review by JasonHowellSlavin — APPROVED** [review]

**Comment by honstar** [comment]
The PR looks good to me 👍 The only thing is the LH score that worries me a bit. @Brandon32, any idea what is causing this?

**Comment by Brandon32** [comment]
@honstar We are loading the external Marketo forms js and this is causing the biggest impact. There are still changes happening on the Marketo side so LH should improve but I'll check on that.

**Comment by honstar** [comment]
@overmyheadandbody @mokimo @narcis-radu : do you think there is a chance to take advantage of the delayed functionality that is being employed for the global-navigation? @Brandon32 : would you agree that marketo forms are typically below the fold? That might give us a chance to delay the 3rd party marketo dependency.

**Comment by mokimo** [comment]
@honstar delaying doesn't really help TBT it's about introducing strategic delays to not block the main thread within the 3rd party scripts, so in this case that would be marketo script providing team making improvements.

Here's what I had shared on slack:
> A few web performance topics I’ve been looking at lately that really helped me to understand TBT and how to avoid it:
https://www.youtube.com/watch?v=7Rrv9qFMWNM
https://web.dev/long-tasks-devtools/
https://web.dev/optimize-long-tasks/
https://philipwalton.com/articles/idle-until-urgent/
https://developer.chrome.com/blog/using-requestidlecallback/
TBT accounts for 30% of the total LH score https://googlechrome.github.io/lighthouse/scorecalc/ - so it really makes sense to get a deep understanding of it
e: Applying the knowledge from above, we broke up a few long running tasks causing TBT in this PR for the global-navigation and it includes a two case studies of the results: https://github.com/adobecom/milo/pull/675 - might be interesting to skim as it is hands on and applied to adobe’ code

**Comment by Brandon32** [comment]
@honstar I would like to give them time to adjust TBT from the Marketo side. I am open to using an intersection observer like we did with charts, most of the time forms are below the fold, but I would want to do that in a separate PR.

File: libs/blocks/marketo/marketo.js
Comment (chrischrischris) [inline libs/blocks/marketo/marketo.js:37]: nit: use destructuring (also shortens line that is > 100)
```suggestion
    const { pathname, search, hash } = destinationUrl;
    destinationUrl = new URL(`${pathname}${search}${hash}`, baseURL.origin);
```

Comment (chrischrischris) [reply libs/blocks/marketo/marketo.js:37]: Looking at code below you prob want to destructure outside the if

File: libs/blocks/marketo/marketo.js
Comment (chrischrischris) [inline libs/blocks/marketo/marketo.js:120]: nit
```suggestion
    if (!key || !value) return;
    if (key in FORM_MAP) {
      formData[FORM_MAP[key]] = value;
    } else {
      formData[key] = value;
    }
```

File: libs/blocks/marketo/marketo.js
Comment (chrischrischris) [inline libs/blocks/marketo/marketo.js:34]: Is destination guaranteed to be a valid url?  If it isn't this will throw

File: libs/blocks/marketo/marketo.js
Comment (Brandon32) [reply libs/blocks/marketo/marketo.js]: Sounds good, thanks!

File: libs/blocks/marketo/marketo.js
Comment (Brandon32) [reply libs/blocks/marketo/marketo.js]: Thanks!

File: libs/blocks/marketo/marketo.js
Comment (Brandon32) [reply libs/blocks/marketo/marketo.js]: Good call, refactored this to a try/catch and made sure we have a hostname, added a test case to validate.

**Comment by mokimo** [comment]
@Brandon32 if it's below the fold then an intersection observer makes a lot of sense 👍

**Pull request merged**
